provider/netcup.go

```diff
@@ -186,7 +190,11 @@ func (p *NetcupProvider) ApplyChanges(ctx context.Context, changes *plan.Changes
 		// Gather records from API to extract the record ID which is necessary for updating/deleting the record
 		recs, err := p.session.InfoDnsRecords(zoneName)
 		if err != nil {
-			_ = level.Error(p.logger).Log("msg", "unable to get DNS records for domain", "zone", zoneName, "error", err)
+			if p.session.LastResponse.Status == string(nc.StatusError) && p.session.LastResponse.StatusCode == 5029 {
```

Comment:
LGTM, i would maybe check additionally for ``if p.session.LastResponse != nil``, just not to assume that my client implementation would always be correct to set this 🤣

Reply:
Good point :D thanks!